synchronization/timer_test.go

```diff
@@ -0,0 +1,79 @@
+package synchronization_test
+
+import (
+	"github.com/orbs-network/orbs-network-go/synchronization"
+	"testing"
+	"time"
+)
+
+func TestTimer_Reset(t *testing.T) {
+	start := time.Now()
+
+	timer := synchronization.NewTimer(10 * time.Millisecond)
```

Comment:
consider using microseconds; if we have a lot of tests sleeping 10ms each, we're dead by a thousand paper cuts

Reply:
correct, becomes flaky at microseconds, but i reduced everything to what seems stable (all the tests now take 10ms to complete